If LLVM Flang is used as the linker driver, it fails on the `-Xlinker` flag.
Pass options to the linker with the `-Wl,` flag instead.

--- libtool-2.5.4/m4/libtool.m4.orig	2025-03-03 15:56:09.384559000 +0100
+++ libtool-2.5.4/m4/libtool.m4	2025-03-03 15:57:39.772092900 +0100
@@ -5264,7 +5264,7 @@
       _LT_TAGVAR(file_list_spec, $1)='@'
 
       if $LD --help 2>&1 | $GREP 'auto-import' > /dev/null; then
-        _LT_TAGVAR(archive_cmds, $1)='$CC -shared $libobjs $deplibs $compiler_flags -o $output_objdir/$soname $wl--enable-auto-image-base -Xlinker --out-implib -Xlinker $lib'
+        _LT_TAGVAR(archive_cmds, $1)='$CC -shared $libobjs $deplibs $compiler_flags -o $output_objdir/$soname $wl--enable-auto-image-base $wl--out-implib,$lib'
 	# If the export-symbols file already is a .def file, use it as
 	# is; otherwise, prepend EXPORTS...
 	_LT_TAGVAR(archive_expsym_cmds, $1)='if _LT_DLL_DEF_P([$export_symbols]); then
